### PR TITLE
feat: export dashboard as json

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

deconstructing #12753

stacked on top of #12959 

## Changes

Adds a button to UI to export dashboards as the dashboard json template

## How did you test this code?

running it locally

### CODE REVIEW

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 1]
```diff
@@ -518,6 +518,40 @@ export const dashboardLogic = kea<dashboardLogicType>({
         ],
     }),
     selectors: () => ({
+        asDashboardTemplate: [
+            (s) => [s.allItems],
+            (dashboard: DashboardType): string => {
+                return dashboard
+                    ? JSON.stringify({
```
Comment (Twixes) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:525]: `JSON.stringify(..., undefined, 4)` for formatting with 4 spaces of indentation ;)

File: frontend/src/scenes/dashboard/dashboardLogic.tsx [hunk 2]
```diff
@@ -526,6 +526,44 @@ export const dashboardLogic = kea<dashboardLogicType>({
...
+                                      return {
+                                          type: 'INSIGHT',
+                                          name: tile.insight.name,
+                                          description: tile.insight.description || '',
+                                          filters: tile.insight.filters,
+                                          layouts: tile.layouts,
+                                          color: tile.color,
+                                      }
+                                  }
+                                  throw new Error('Unknown tile type')
```
Comment (mariusandra) [inline frontend/src/scenes/dashboard/dashboardLogic.tsx:558]: Will this take down the entire app? 💥

File: frontend/src/types.ts [hunk 3]
```diff
@@ -2309,21 +2309,32 @@ export enum ExporterFormat {
     PNG = 'image/png',
     CSV = 'text/csv',
     PDF = 'application/pdf',
+    JSON = 'application/json',
 }
 
+export type LocalExportContext = {
+    localData: string // when exporting directly from the browser to a file
```
Comment (mariusandra) [inline frontend/src/types.ts:2316]:
```suggestion
/** Exporting directly from the browser to a file */
export type LocalExportContext = {
    localData: string
```

**Review by mariusandra — APPROVED** [review]
Does what it says on the label 👍.

Left a suggestions, ignore/merge as you wish.

File: frontend/src/scenes/dashboard/dashboardLogic.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/dashboard/dashboardLogic.tsx:558]: Fair, move of an error toast scenario... will sweep up in follow up

**Pull request merged**